### PR TITLE
Deprecate SbFileOpen and SbFileClose.

### PR DESCRIPTION
**Pull request by yjzhang111**
b/302715109

### CODE REVIEW

File: starboard/shared/win32/file_close.cc [hunk 1]
```diff
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// #if SB_API_VERSION < 16
```
Comment (y4vor) [inline starboard/shared/win32/file_close.cc:15]: too early for deprecation, please remove the commeent.

File: third_party/musl/src/starboard/fcntl/open.c [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#if SB_API_VERSION < 16
+
+#include <errno.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+
+#include "starboard/file.h"
+#include "starboard/shared/posix/file_internal.h"
```
Comment (y4vor) [inline third_party/musl/src/starboard/fcntl/open.c:8]: We can't use Starboard internal implementation outside of Starboard.
I think you might need to emulate file descriptors similar to what Max did for sockets.

**Review by y4vor — COMMENTED** [review]
I think for win32 the open would have to coordinate with the sb_close.
starboard/shared/win32/posix_emu/socket.cc

File: starboard/shared/win32/file_close.cc (hunk 1 above)
Comment (yjzhang111) [reply starboard/shared/win32/file_close.cc]: Removed.

File: third_party/musl/src/starboard/fcntl/open.c (hunk 2 above)
Comment (yjzhang111) [reply third_party/musl/src/starboard/fcntl/open.c]: Removed.

File: starboard/shared/win32/posix_emu/socket.cc [hunk 3]
```diff
@@ -94,22 +103,45 @@ int sb_socket(int domain, int type, int protocol) {
...
 }
 
 int close(int fd) {
-  SOCKET socket_handle = handle_db_get(fd, true);
+  FileOrSocket handle = handle_db_get(fd, true);
 
-  if (socket_handle != INVALID_SOCKET) {
-    return closesocket(socket_handle);
+  if (!handle.is_file) {
+    return closesocket(handle.socket);
```
Comment (y4vor) [inline starboard/shared/win32/posix_emu/socket.cc:136]: Check INVALID_SOCKET?

Comment (yjzhang111) [reply starboard/shared/win32/posix_emu/socket.cc:139]: Done.

**Review by y4vor — APPROVED** [review]

**Pull request merged**